server/src/main/java/org/opensearch/repositories/RepositoriesService.java

```diff
@@ -288,6 +317,7 @@ public ClusterState execute(ClusterState currentState) {
                         boolean changed = false;
                         for (RepositoryMetadata repositoryMetadata : repositories.repositories()) {
                             if (Regex.simpleMatch(request.name(), repositoryMetadata.name())) {
+                                ensureNonSystemRepository(repositoryMetadata.settings(), request.name());
```

Comment:
The error message would be confusing here : `cannot register a system repository externally` . We need to modify it .